vocab-builder-cli/src/main/java/com/github/tkurz/sesame/vocab/Main.java

```diff
@@ -100,38 +100,82 @@ public static void main(String[] args) {
             if (cli.hasOption('l')) {
                 builder.setPreferredLanguage(cli.getOptionValue('l'));
             }
-            if (cli.hasOption('S')) {
-                builder.setStringPropertySuffix(cli.getOptionValue('S'));
-            } else {
-                builder.setStringPropertySuffix(null);
-            }
-            if (cli.hasOption('P')) {
-                builder.setStringPropertyPrefix(cli.getOptionValue('P'));
-            } else {
-                builder.setStringPropertyPrefix(null);
-            }
+            CaseFormat caseFormat = null;
+            final GenerationSetting uriGenerationSetting = new GenerationSetting(); 
             if (cli.hasOption('c')) {
+            	String uriCaseFormat = cli.getOptionValue('c');
+            	uriGenerationSetting.setCaseFormat(uriCaseFormat);
```

Comment:
This is an odd style of programming to optimistically set the case format and then check it later and throw an exception. Typically, to make code maintainable, the setter is only called after it is validated (or the validation happens inside of the setter).